Source/WebCore/workers/service/ServiceWorkerGlobalScope.cpp

```diff
@@ -271,4 +276,85 @@ CookieStore& ServiceWorkerGlobalScope::cookieStore()
     return *m_cookieStore;
 }
 
+void ServiceWorkerGlobalScope::addFetchTask(FetchKey key, Ref<ServiceWorkerFetch::Client>&& client)
+{
+    ASSERT(!m_ongoingFetchTasks.contains(key));
+    m_ongoingFetchTasks.add(key, FetchTask { WTFMove(client), nullptr });
+}
+
+void ServiceWorkerGlobalScope::addFetchEvent(FetchKey key, FetchEvent& event)
+{
+    ASSERT(m_ongoingFetchTasks.contains(key));
+    auto iterator = m_ongoingFetchTasks.find(key);
```

Comment:
`ASSERT(iterator != m_ongoingFetchTasks.end());` to avoid double lookup?

Reply:
The ASSERT(contains) is more readable.